src/emulator/modules/SceNet/src/SceNet.cpp

```diff
@@ -105,60 +209,102 @@ EXPORT(int, sceNetGetpeername) {
     return unimplemented("sceNetGetpeername");
 }
 
-EXPORT(int, sceNetGetsockname) {
-    return unimplemented("sceNetGetsockname");
+EXPORT(int, sceNetGetsockname, int s, SceNetSockaddr *name, unsigned int *namelen) {
+    int res = get_socket_address(host.net, s, name, namelen);
+    if (res < 0){
+        return error("sceNetGetsockname", translate_errorcode());   
+    }else{
+        return res;
+    }
 }
 
 EXPORT(int, sceNetGetsockopt) {
     return unimplemented("sceNetGetsockopt");
 }
 
-EXPORT(int, sceNetHtonl) {
-    return unimplemented("sceNetHtonl");
+EXPORT(unsigned int, sceNetHtonl, unsigned int n) {
+    return htonl(n);
 }
 
 EXPORT(int, sceNetHtonll) {
     return unimplemented("sceNetHtonll");
 }
 
-EXPORT(int, sceNetHtons) {
-    return unimplemented("sceNetHtons");
+EXPORT(unsigned short int, sceNetHtons, unsigned short int n) {
+    return htons(n);
 }
 
-EXPORT(int, sceNetInetNtop) {
-    return unimplemented("sceNetInetNtop");
+EXPORT(Ptr<const char>, sceNetInetNtop, int af, const void *src, Ptr<char> dst, unsigned int size) {
+    char *dst_ptr = dst.get(host.mem);
+#ifdef WIN32
+    const char *res = InetNtop(af, src, dst_ptr, size);
+#else
+    const char *res = inet_ntop(af, src, dst_ptr, size);
+#endif
+    if (res == nullptr){
+        return error("sceNetInetNtop", translate_errorcode()); 
```

Comment:
This needs to return a `Ptr<char>()` (null pointer in Ptr language) rather than translating the error code.